File: Player/src/main/java/ru/testsimpleapps/coloraudioplayer/managers/tools/PreferenceTool.java
package ru.testsimpleapps.coloraudioplayer.managers.tools;

import android.content.Context;
import android.content.SharedPreferences;
import android.support.annotation.NonNull;

import ru.testsimpleapps.coloraudioplayer.App;
import ru.testsimpleapps.coloraudioplayer.managers.explorer.Data.ConfigData;


public class PreferenceTool {

    public static final String TAG = PreferenceTool.class.getSimpleName();

    /*
    * Keys
    * */
    private final String KEY_CONTROL_IS_EXPAND = "KEY_CONTROL_IS_EXPAND";
    private final String KEY_EXPLORER_GROUP = "KEY_EXPLORER_GROUP";
    private final String KEY_EXPLORER_SORT = "KEY_EXPLORER_SORT";
    private final String KEY_EXPLORER_SORT_ORDER = "KEY_EXPLORER_SORT_ORDER";
    private final String KEY_CONTROL_PANEL = "KEY_CONTROL_PANEL";
    private final String KEY_CONTROL_INFO = "KEY_CONTROL_INFO";
    private final String KEY_PLAYLIST_EXPAND = "KEY_PLAYLIST_EXPAND";

    private static volatile PreferenceTool sPreferenceTool;
    private SharedPreferences mSharedPreferences;

    private PreferenceTool(@NonNull Context context) {
        mSharedPreferences = context.getSharedPreferences(TAG, context.MODE_PRIVATE);
    }

    public static PreferenceTool getInstance() {
        PreferenceTool preferenceTool = sPreferenceTool;
        if (preferenceTool == null) {
            synchronized (PreferenceTool.class) {
                preferenceTool = sPreferenceTool;
                if (preferenceTool == null) {
                    sPreferenceTool = preferenceTool = new PreferenceTool(App.getContext());
                }
            }
        }

        return preferenceTool;
    }

    public boolean getControlPanelExpand() {
        return mSharedPreferences.getBoolean(KEY_CONTROL_IS_EXPAND, false);
    }

    public void setControlPanelExpand(boolean value) {
        mSharedPreferences.edit().putBoolean(KEY_CONTROL_IS_EXPAND, value).commit();
    }

    public void setExplorerGroupType(final int value) {
        mSharedPreferences.edit().putInt(KEY_EXPLORER_GROUP, value).commit();
    }

    public int getExplorerGroupType() {
        return mSharedPreferences.getInt(KEY_EXPLORER_GROUP, ConfigData.GROUP_TYPE_FOLDERS);
    }

    public void setExplorerSortType(final int value) {
        mSharedPreferences.edit().putInt(KEY_EXPLORER_SORT, value).commit();
    }

    public int getExplorerSortType() {
        return mSharedPreferences.getInt(KEY_EXPLORER_SORT, ConfigData.SORT_TYPE_NAME);
    }

    public void setExplorerSortOrder(final int value) {
        mSharedPreferences.edit().putInt(KEY_EXPLORER_SORT_ORDER, value).commit();
    }

    public int getExplorerSortOrder() {
        return mSharedPreferences.getInt(KEY_EXPLORER_SORT_ORDER, ConfigData.SORT_ORDER_ASC);
    }

    public void setControlPanel(final boolean value) {
        mSharedPreferences.edit().putBoolean(KEY_CONTROL_PANEL, value).commit();
    }

    public boolean getControlPanel() {
        return mSharedPreferences.getBoolean(KEY_CONTROL_PANEL, true);
    }

    public void setControlInfo(final boolean value) {
        mSharedPreferences.edit().putBoolean(KEY_CONTROL_INFO, value).commit();
    }

    public boolean getControlInfo() {
        return mSharedPreferences.getBoolean(KEY_CONTROL_INFO, true);
    }

    public boolean getPlaylistViewExpand() {
        return mSharedPreferences.getBoolean(KEY_PLAYLIST_EXPAND, true);
    }

    public void setPlaylistViewExpand(final boolean value) {
        mSharedPreferences.edit().putBoolean(KEY_PLAYLIST_EXPAND, value);
    }

}

File: player/src/main/java/ru/testsimpleapps/coloraudioplayer/managers/tools/PreferenceTool.kt
package ru.testsimpleapps.coloraudioplayer.managers.tools

import android.content.Context
import android.content.SharedPreferences

import ru.testsimpleapps.coloraudioplayer.app.App
import ru.testsimpleapps.coloraudioplayer.managers.explorer.Data.ConfigData


class PreferenceTool private constructor(context: Context) {

    /*
    * Keys
    * */
    private val KEY_CONTROL_IS_EXPAND = "KEY_CONTROL_IS_EXPAND"
    private val KEY_EXPLORER_GROUP = "KEY_EXPLORER_GROUP"
    private val KEY_EXPLORER_SORT = "KEY_EXPLORER_SORT"
    private val KEY_EXPLORER_SORT_ORDER = "KEY_EXPLORER_SORT_ORDER"
    private val KEY_CONTROL_PANEL = "KEY_CONTROL_PANEL"
    private val KEY_CONTROL_INFO = "KEY_CONTROL_INFO"
    private val KEY_PLAYLIST_EXPAND = "KEY_PLAYLIST_EXPAND"
    private val mSharedPreferences: SharedPreferences

    var controlPanelExpand: Boolean
        get() = mSharedPreferences.getBoolean(KEY_CONTROL_IS_EXPAND, false)
        set(value) {
            mSharedPreferences.edit().putBoolean(KEY_CONTROL_IS_EXPAND, value).commit()
        }

    var explorerGroupType: Int
        get() = mSharedPreferences.getInt(KEY_EXPLORER_GROUP, ConfigData.GROUP_TYPE_FOLDERS)
        set(value) {
            mSharedPreferences.edit().putInt(KEY_EXPLORER_GROUP, value).commit()
        }

    var explorerSortType: Int
        get() = mSharedPreferences.getInt(KEY_EXPLORER_SORT, ConfigData.SORT_TYPE_NAME)
        set(value) {
            mSharedPreferences.edit().putInt(KEY_EXPLORER_SORT, value).commit()
        }

    var explorerSortOrder: Int
        get() = mSharedPreferences.getInt(KEY_EXPLORER_SORT_ORDER, ConfigData.SORT_ORDER_ASC)
        set(value) {
            mSharedPreferences.edit().putInt(KEY_EXPLORER_SORT_ORDER, value).commit()
        }

    var controlPanel: Boolean
        get() = mSharedPreferences.getBoolean(KEY_CONTROL_PANEL, true)
        set(value) {
            mSharedPreferences.edit().putBoolean(KEY_CONTROL_PANEL, value).commit()
        }

    var controlInfo: Boolean
        get() = mSharedPreferences.getBoolean(KEY_CONTROL_INFO, true)
        set(value) {
            mSharedPreferences.edit().putBoolean(KEY_CONTROL_INFO, value).commit()
        }

    var playlistViewExpand: Boolean
        get() = mSharedPreferences.getBoolean(KEY_PLAYLIST_EXPAND, true)
        set(value) {
            mSharedPreferences.edit().putBoolean(KEY_PLAYLIST_EXPAND, value).commit()
        }

    init {
        mSharedPreferences = context.getSharedPreferences(TAG, Context.MODE_PRIVATE)
    }

    companion object {

        val TAG = PreferenceTool::class.java.simpleName

        @Volatile
        private var sPreferenceTool: PreferenceTool? = null

        val instance: PreferenceTool
            get() {
                var preferenceTool = sPreferenceTool
                if (preferenceTool == null) {
                    synchronized(PreferenceTool::class.java) {
                        preferenceTool = sPreferenceTool
                        if (preferenceTool == null) {
                            preferenceTool = PreferenceTool(App.instance)
                            sPreferenceTool = preferenceTool
                        }
                    }
                }

                return preferenceTool!!
            }
    }

}
